pragma-node/src/infra/repositories/onchain_repository.rs

```diff
@@ -244,38 +348,53 @@ pub async fn routing(
             .await?;
             let alt_quote_decimal = get_decimals(offchain_pool, &alt_quote_pair).await?;
 
-            let rebased_price = calculate_rebased_price(
-                (base_alt_result.0, base_alt_decimal),
-                (alt_quote_result.0, alt_quote_decimal),
-            )?;
-            base_alt_result.1.extend(alt_quote_result.1);
-            return Ok(RawOnchainData {
-                price: rebased_price.0,
-                decimal: rebased_price.1,
-                sources: base_alt_result.1,
-                pair_used: vec![base_alt_pair, alt_quote_pair],
-            });
+            if alt_quote_result.len() != base_alt_result.len() {
+                return Err(InfraError::InternalServerError);
+            }
+            let mut result: Vec<RawOnchainData> = Vec::new();
```

Comment:
This deserve its own function, we're doing a lot of different things here

Reply:
done